sequence_processing_pipeline/ConvertJob.py

```diff
@@ -210,3 +218,158 @@ def parse_job_script(job_script_path):
+                    if re.match(rgx, file_name):
+                        sample['matching_files'].append(full_path)
+
+        self.info = results
+
+    def copy_sequences(self, sample_name, source_project, dest_project,
+                       copy_all_replicates=False):
+        """
+        Copies all fastq files related to a sample into another project.
+        :param source_project: The source project w/qiita_id.
+        :param dest_project: The destination project w/qiita_id.
+        :param sample_name: A sample-name.
+        :param orig_name: A sample-name.
+        :param copy_all_replicates: If True, search for sample_name in the
+            orig_name column of the sample-sheet. Copy all replicates.
+        :return: None
+        """
+        if self.info is None:
+            raise ValueError("This method cannot be called until processing "
+                             "has completed.")
+
+        project_names = list(self.info.keys())
+
+        # confirm source project is a valid one.
+        if source_project not in project_names:
+            raise ValueError(f"'{source_project}' is not defined in the "
+                             "sample-sheet")
+
+        # confirm destination project is a valid one.
+        if dest_project not in project_names:
+            raise ValueError(f"'{dest_project}' is not defined in the "
+                             "sample-sheet")
+
+        if source_project == dest_project:
+            raise ValueError(f"source '{source_project}' and destination "
+                             f"'{dest_project}' projects are the same")
+
+        # note that the user can supply a sample-name that didn't make it
+        # through the conversion step and may have no files matched to it.
+        # this is considered okay and a possible outcome of conversion. In
+        # this case zero files are copied and no information is passed back
+        # to the user.
+
+        # projects that contain replicates must also be considered. if the
+        # value for sample_name contains a well-id, then only the files
+        # associated with that particular replicate should be copied. If the
+        # value instead references a sample_name in the 'orig_name' column,
+        # then all files associated with each replicate need to be moved.
+
+        # in this situation, the sample_name needs be compared against all
+        # orig_names in the project and the individual sample_names (w/well-
+        # ids) must be discovered. Then those individual sample_names can
+        # be processed.
+
+        if copy_all_replicates is True and self.contains_replicates is False:
+            raise ValueError("'treat_as_orig_name' is set to 'True' but this "
+                             "sample-sheet doesn't contain replicates")
+
+        samples = self.info[source_project]['samples']
+        sample_list = []
+
+        if copy_all_replicates:
+            for key in samples:
```

Comment:
If we don't decide to take the approach suggested above, how about replacing 351-352 with:

`for _, sample in samples.items():`

since it doesn't look like `key` is being in this scope used except to get `sample`?